bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/rest/GsonMapper.java

```diff
@@ -329,6 +460,41 @@ public Optional<Object> datapointValue(@Nullable String json) {
         return Optional.ofNullable(datapoint.get("value"));
     }
 
+    public List<String> parseAwsGatewayIds(String json) {
+        var map = tryParseBody(json, MAP_TYPE_REFERENCE, Map.of());
+        if (!map.containsKey("data")) {
+            return List.of();
+        }
+
+        var data = map.get("data");
+        List<Object> list;
+        if (data instanceof Collection<?> collection) {
+            list = new ArrayList<>(collection);
+        } else {
+            list = tryParseBody(data.toString(), LIST_TYPE_REFERENCE, List.of());
+        }
+
+        return list.stream()//
+                .map(this::parseAwsGatewayId)//
+                .filter(Optional::isPresent)//
+                .map(Optional::get)//
+                .toList();
+    }
+
+    @SuppressWarnings("unchecked")
```

Comment:
Can you get rid of the suppression?

Reply:
https://github.com/openhab/openhab-addons/pull/16807#discussion_r1616363461